ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -27,8 +27,257 @@
+
+        // Delete the uploaded file.
+        client.DeleteObject(test_bucket_name, test_bucket_prefix + object_name);
+    }
+
+    SECTION("Simple delete test", "[gcp-connection]")
+    {
+
+        // Delete a file that does not exist in the bucket - should fail.
+        REQUIRE(conn.delete_object(non_existant_object_name) == ENOENT);
+
+        // Upload a test file.
+        REQUIRE(
+          upload_file(client, test_bucket_name, test_bucket_prefix, file_name, object_name).ok());
+        REQUIRE(file_exists_in_bucket(client, test_bucket_name, test_bucket_prefix, object_name));
+
+        // Delete the uploaded file.
+        REQUIRE(conn.delete_object(object_name) == 0);
+
+        // Check that the file has been deleted.
+        REQUIRE_FALSE(
+          file_exists_in_bucket(client, test_bucket_name, test_bucket_prefix, object_name));
+    }
+
+    SECTION("Simple object exists test", "[gcp-connection]")
+    {
+        bool exists;
+        size_t size;
+
+        REQUIRE(conn.object_exists(object_name, exists, size) == 0);
+        REQUIRE(exists == false);
+        REQUIRE(size == 0);
+
+        // Upload a test file.
+        REQUIRE(
+          upload_file(client, test_bucket_name, test_bucket_prefix, file_name, object_name).ok());
+        REQUIRE(num_objects_in_bucket(client, test_bucket_name, test_bucket_prefix) == 1);
+
+        // Check the bucket contains the uploaded file.
+        REQUIRE(conn.object_exists(object_name, exists, size) == 0);
+        REQUIRE(exists);
+        REQUIRE(size != 0);
+
+        // Delete the uploaded file.
+        auto dl_metadata = client.DeleteObject(test_bucket_name, test_bucket_prefix + object_name);
+        REQUIRE(dl_metadata.ok());
+
+        // Check that the file has been deleted.
+        REQUIRE_FALSE(
+          file_exists_in_bucket(client, test_bucket_name, test_bucket_prefix, object_name));
+
+        // Check for a file that is not in the bucket, object exists should modify the exists
+        // variable to false.
+        REQUIRE(conn.object_exists(object_name, exists, size) == 0);
+        REQUIRE(exists == false);
+        REQUIRE(size == 0);
+    }
+
+    SECTION("Read tests", "[gcp-connection]")
+    {
+        // Upload a test file.
+        REQUIRE(
+          upload_file(client, test_bucket_name, test_bucket_prefix, file_name, object_name).ok());
+        REQUIRE(file_exists_in_bucket(client, test_bucket_name, test_bucket_prefix, object_name));
+
+        // Read GCP objects under the test bucket with no offset.
+        char buf[1024];
+
+        REQUIRE(conn.read_object(object_name, 0, payload.length(), buf) == 0);
+        REQUIRE(payload.compare(buf) == 0);
+        memset(buf, 0, 1000);
+
+        // Read GCP objects under the test bucket with offset.
+        const int str_len = payload.length() - payload.find(" ");
+        REQUIRE(conn.read_object(object_name, payload.find(" "), str_len, buf) == 0);
+        REQUIRE(payload.substr(4, 16).compare(buf) == 0);
```

Comment:
are these numbers still meant to be here after the latest commit?

Reply:
fixed [134ac50](https://github.com/wiredtiger/wiredtiger/pull/8749/commits/134ac50062003f4cc1643507f6b61f40adef6754)